frontend/public/module/k8s/k8s-actions.js

```diff
@@ -139,8 +148,23 @@ const actions = {
      */
     const pollAndWatch = () => (delete POLLs[id]) && incrementallyLoad()
       .then(resourceVersion => {
+        // ensure this watch should still exist because pollAndWatch is recursiveish
+        if (!REF_COUNTS[id]) {
+          // eslint-disable-next-line no-console
+          console.log(`stopped watching ${id} before finishing incremental loading.`);
+          // call cleanup function out of abundance of caution...
+          dispatch(actions.stopK8sWatch(id));
+          return;
+        }
+
+        if (WS[id]) {
+          // eslint-disable-next-line no-console
+          console.warn(`Attempted to create multiple websockets for ${id}.  This should never happen.`);
```

Comment:
Should this really never happen? I figured there had to be some page with two components that watch the same thing.

Reply:
Good question!  I think the answer is yes, but tell me if this makes sense:

`watchK8sList` checks REF_COUNTS first thing to see if something else is watching the same `id` and NOPs out if so:

```javascript
  watchK8sList: (id, query, k8skind) => (dispatch, getState) => {
    // Only one watch per unique list ID
    if (id in REF_COUNTS) {
      REF_COUNTS[id] += 1;
      return nop;
    }
```

List components get their data out of Redux by `id`, so we only ever need one WS and one set of callbacks updating the store.  If we don't exit early, we potentially dispatch the same events multiple times (that may work in practice, or it may not, who knows), but I think the only way we hit that condition is if `WS[id]` exists and `REF_COUNTS[id]` does not ... which was only possible because of the race condition fixed in this PR.